Inventory.cs

```diff
@@ -2,30 +2,29 @@
 
 public class Inventory
 {
-    private List<Product> _products = new();
-
-    public void AddProduct(Product newProduct)
+    public async Task AddProductAsync(Product newProduct)
     {
-        _products.Add(newProduct);
+        await MongoDb.Instance.AddProductAsync(newProduct);
```

Comment:
seems like here and in the sqlserver MR you trying to update the same Invetory class to use a different DB, which is not ideal. Imagine you started your project with SQL server and 10+ DB repo classes, and then you wanted to switch to MongoDB, then you would need to update all the 10+ classes to use the new db. Here, we would need to apply liskov substitution principle, where we have one interface for IProductRepository, and two different implementations (one SQL Server, the other one MongoDB). Then in this inventory class, you would be dealing with the interface instead, and pass that from the constructor, and the consumer of this class can pass either the MongoDB implementation or SQL Server.

```csharp
public class Inventory
{
public Inventory(IProductRepository repo) {}
}

public class Main
{
// You would use this
var inventory = new Inventory(new SqlServerProductRepositry());
// Or switch later to this
var inventory = new Inventory(new MongoDbProductRepositry());
}
```